prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -237,4 +239,23 @@ class FHIRTranslator(
         }
         return this
     }
+}
+
+/**
+ * Encodes a message while avoiding an error when MSH-2 is five characters long
+ *
+ * @return the encoded message as a string
+ */
+fun Message.encodePreserveEncodingChars(): String {
```

Comment:
We discussed making this function more generic and having it take in params, but agreed there does not seem to be another clear use case for this. We can revisit this idea in the future if it comes up.